bika/lims/browser/bika_listing.py

```diff
@@ -1257,11 +1257,24 @@ def _folderitems(self, full_objects=False):
                 after={},
                 replace={},
             )
-            rs = self.workflow.getInfoFor(obj, 'review_state')
-            st_title = self.workflow.getTitleForStateOnType(rs, obj.portal_type)
-            st_title = t(PMF(st_title))
+
+            rs = None
+            wf_state_var = None
+
+            workflows = self.workflow.getWorkflowsFor(obj)
+            for wf in workflows:
+                if wf.state_var:
+                    wf_state_var = wf.state_var
```

Comment:
This is really nice! When we merge senaite-integration to master, most secondary workflows (notably cancellation and deactivation) will be removed, and review_state will be used to manage these states.